lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -443,51 +445,82 @@ def _mtab_entries(self):
             mtab_entries.append(fields)
         return mtab_entries
 
-    @timeout.timeout()
+    def get_mount_info(self, mount_info, bind_mounts, uuids):
+
+            mount_info.update(get_mount_size(mount_info['mount']))
```

Comment:
Don't modify things inside of a function unless there's a huge performance hit.  Just return the values.

Also, don't pass in a meta object like mount_info...  Instead be explicit that you need the mount and the device by passing just those two arguments in.

Looking at this further, though, it probably makes more sense to call each of those separately as they're disjoint information and could timeout separately.

Reply:
refactored as per comments, keeping both operations in same thread for now, since if the mount is going to be problematic, both of those would trigger the same problem. 

Also i would abstract the threading a bit vs having to code it for every possible function that can cause stalls/timeouts. First i want to see if this actually does improve those situations though.